core/file.go

```diff
@@ -18,9 +18,16 @@ import (
 	"github.com/moby/buildkit/solver/pb"
 	"github.com/opencontainers/go-digest"
 	specs "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
 	fstypes "github.com/tonistiigi/fsutil/types"
 )
 
+var (
+	// ErrFileContentsExceedsLimit is returned when a file that exceeds
+	// MaxFileContentsSize is passed to Contents method:
+	ErrFileContentsExceedsLimit = errors.New("file exceeds limit")
```

Comment:
nit: we could also include the value of the limit and the actual size in the error message too, which would be helpful in case anyone actually hits this